### DISCUSSION TITLE
What does "Only download images for BoilR shortcuts" mean?

### DISCUSSION THREAD

**Discussion by extremin** [PhilipK/BoilR#335] [Q&A]
In the Settings menu tab, there is the checkbox option says "Only download images for BoilR shortcuts" under SteamGridDB.
I don't quite understand the meaning of this option.
What is even "BoilR shortcuts" exactly?

I haven't recognized any difference whether I checked it or unchecked it.
More detailed explaination will be very appreciated.

<img width="737" alt="boilr option" src="https://user-images.githubusercontent.com/54126648/220418643-89be25d4-ab70-4a55-a79e-043914d2226d.png">

**Comment by Charlie284** [comment]
A “BoilR shortcut” is a non-Steam shortcut that BoilR imported into Steam from one of the launchers/platforms it scans. It is not every shortcut in Steam, and it does not mean native Steam games.

BoilR marks the shortcuts it manages with the value `boilr` (historically in the shortcut's `DevkitGameID`, with the `boilr` tag also recognized). The image downloader reads Steam's non-Steam shortcuts and applies this filter:

```text
option off: every non-Steam shortcut may be considered
option on:  only shortcuts carrying BoilR's marker are considered
```

That is visible in the source from the time of the question:

- Imported shortcuts receive the `boilr` marker [during synchronization](https://github.com/PhilipK/BoilR/blob/03f86639537f45e5a29ed3f2b8bd282ebf51a7d3/src/sync/synchronization.rs#L52-L66).
- A shortcut is classified as BoilR-managed by its tag or `DevkitGameID` marker [here](https://github.com/PhilipK/BoilR/blob/03f86639537f45e5a29ed3f2b8bd282ebf51a7d3/src/sync/synchronization.rs#L138-L146).
- The SteamGridDB search skips unmarked shortcuts only when this setting is enabled [here](https://github.com/PhilipK/BoilR/blob/03f86639537f45e5a29ed3f2b8bd282ebf51a7d3/src/steamgriddb/downloader.rs#L165-L198).

For example, suppose Steam's non-Steam library contains:

1. an Epic game imported by BoilR;
2. an emulator shortcut you added manually;
3. a local script you added manually.

With the box checked, automatic/bulk SteamGridDB artwork lookup considers only item 1. With it unchecked, it can also search for missing artwork for items 2 and 3.

The setting does not change which games BoilR imports, and it does not remove existing artwork. It only changes which non-Steam shortcuts are candidates when an image download runs. You may therefore see no visible difference if all your non-Steam shortcuts came from BoilR, the other shortcuts already have all requested artwork, or SteamGridDB cannot match their names.
